ID3FrameStringContentParser crash fix

Scytalion opened the pull request:
Fixes ID3FrameStringContentParser crash with invalid range

## Description
Added a guard before data is accessed.

## Motivation and Context
Some of my mp3 files are running into this crash.

## How Has This Been Tested?
Added a unit test.

## Types of changes
- [x] Bug fix :bug: (non-breaking change which fixes an issue)

## Checklist:
- [x] My code follows the code style of this project :beers:.
- [x] I have read the [CONTRIBUTING](https://github.com/chicio/ID3TagEditor/blob/master/CONTRIBUTING.md) document :busts_in_silhouette:.
- [x] I have added tests to cover my changes :tada:.
- [x] All new and existing tests passed :white_check_mark:.

chicio commented:
Hi @Scytalion and thank you so much for the contribution 🙏. I will review the PR in the next days and I will merge it if I don't have any comments.
Also, I will add you to the contributors section 🚀🚀

chicio commented:
Ps please @Scytalion support my work by starring ⭐️ this repo ⭐️ (if you still didn't do it)

chicio commented:
Seems good to me. I will merge and release a new version by the end of the week. Thank you so much again for the contribution 🙏

chicio commented:
ps there are some listing problem. If you have time fix them (or i will fix before/after merge)

Scytalion commented:
Yes of course, no problem. I had no SwiftLint installed, until now :-). I moved the SwiftLint build phase into an extra script and added a fix for apple silicon SwiftLint brew installs.

chicio commented:
Version 4.6.0 released on cocoapods and for SPM 🚀
